client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java

```diff
@@ -60,21 +66,31 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
         notificationBuilder.setContentIntent(pendingIntent);
         NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
         notificationManager.notify(0,notificationBuilder.build());
+        Notify(remoteMessage.getData());
     }
 
 
+
     @Override
     public void attach(ActivityObserver observerToAdd) {
 
+        if(observers.contains(observerToAdd)){
+            return;
+        }
+        observers.add(observerToAdd);
     }
 
     @Override
     public void detach(ActivityObserver observerToRemove) {
-
+        observers.remove(observerToRemove);
     }
 
     @Override
-    public void Notify(String response) {
+    public void Notify() {
+
+        for(ActivityObserver e : observers){
```

Comment:
There should be a space between the closing parenthesis and opening curly bracket.